includes/functions.php

```diff
@@ -118,12 +119,14 @@ function generate_toc($html_string){
       $name = trim(str_replace('&nbsp;','', htmlentities(strip_tags($h_content) )));
       if($level > $curr_level){
         $toc .= "\n".'<div class="list-group">'."\n";
+        $counter +=1;
```

Comment:
I wonder if this also needs to have a `while` loop to mirror the one for closing divs?
```suggestion
        while($level < $curr_level){
          $toc .= "\n".'<div class="list-group">'."\n";
          $curr_level += 1;
          $counter +=1;
        }
```

Reply:
No. This is what I try to say with the empty list-group above. This will not have an `<a>`-tag, which would be able to have an `active` class. So we could not expand the corresponding list-group sibling in the css.

Reply:
ok, so the ToC hierarchy would look like h1 h2 even though it's h1 h3 I guess?

In fairness this shouldn't really happen. I think that one of the markdownlint looks for non-consecutive headings - not sure if we have turned that off or not though.

Reply:
yes, but I don't think it will have an effect on the rendering of the ToC. 
Top-level params would be h3 without a parent h2 element.

Reply:
ok even better!